index.js

```diff
@@ -11,44 +12,119 @@ function constructRequest(har, userAgent = false) {
+          // within a browser environment you'll have two major differences in API:
+          //
+          //  * The `.append()` API in `form-data` requires that the third argument is an object containing various,
+          //    undocumented, options. In the browser, `.append()`'s third argument should only be present when the
+          //    second is a `Blob` or `USVString`, and when it is present, it should be a filename string.
+          //  * `form-data` does not expose an `.entries()` API, so the only way to retrieve data out of it for
+          //    construction of boundary-separated payload content is to use its `.pipe()` API. Since the browser
+          //    doesn't have this API, you'll be unable to retrieve data out of it.
+          //
+          // Now since the native `FormData` API is iterable, and has the `.entries()` iterator, we can easily detect
+          // what version of the FormData API we have access to by looking for this and constructing a simple wrapper
+          // to disconnect some of this logic so you can work against a single, consistent API.
+          //
+          // Having to do this isn't fun, but it's the only way you can write code to work with `multipart/form-data`
+          // content under a server and browser.
+          const form = new FormData();
+          const isNativeFormData = typeof form[Symbol.iterator] === 'function';
+
+          request.postData.params.forEach(param => {
+            if ('fileName' in param && !('value' in param)) {
+              throw new Error(
+                "The supplied HAR has a postData parameter with `fileName`, but no `value` content. Since this library doesn't have access to the filesystem, it can't fetch that file."
+              );
+            }
+
+            if (isNativeFormData) {
+              if ('fileName' in param) {
+                const paramBlob = new Blob([param.value], { type: param.contentType || null });
+                form.append(param.name, paramBlob, param.fileName);
+              } else {
+                form.append(param.name, param.value);
+              }
+            } else {
+              form.append(param.name, param.value || '', {
+                filename: param.fileName || null,
+                contentType: param.contentType || null,
+              });
+            }
+          });
+
+          options.body = form;
+          break;
+
+        default:
+          const formBody = {};
+          request.postData.params.map(param => {
+            try {
+              formBody[param.name] = JSON.parse(param.value);
+            } catch (e) {
+              formBody[param.name] = param.value;
+            }
+
+            return true;
+          });
+
+          options.body = JSON.stringify(formBody);
       }
     } else {
       options.body = request.postData.text;
     }
   }
 
-  if (request.queryString.length) {
+  if ('queryString' in request && request.queryString.length) {
```

Comment:
Wonder if it's worth taking a look at optional chaining now. I know I've said I don't like it, but it would really clean up a lot of these types of conditional statements 😬

Reply:
I'm with you there. I don't really care for the obtuse syntax of optional chaining, but having code this verbose everywhere is a bit tedious.